static/src/javascripts/projects/common/modules/identity/consents.js

```diff
@@ -241,38 +238,51 @@ const bindCheckAllSwitch = (labelEl: HTMLElement): void => {
         fetchWrappedCheckboxes(),
         bindCheckboxAnalyticsEventsOnce(labelEl),
     ]).then(([[checkboxEl, titleEl], wrappedCheckboxEls]) => {
+        if (!(checkboxEl instanceof HTMLInputElement)) {
+            throw new Error(ERR_MALFORMED_HTML);
+        }
+
         const getTextForStatus = (status: boolean) =>
             status ? LC_UNCHECK_ALL : LC_CHECK_ALL;
 
         const updateCheckStatus = () =>
             fastdom.write(() => {
-                if (!(checkboxEl instanceof HTMLInputElement)) {
-                    throw new Error(ERR_MALFORMED_HTML);
-                }
                 checkboxEl.checked = getCheckedAllStatus(wrappedCheckboxEls);
                 titleEl.innerHTML = getTextForStatus(checkboxEl.checked);
                 labelEl.style.visibility = 'visible';
                 labelEl.style.pointerEvents = 'all';
             });
 
+        const getInputFieldsFromCheckboxes = checkboxesToUpdate =>
```

Comment:
Not sure about this method name.  It's actually retrieving an array of promises, which when complete will contain an array (node list) of input fields.